Makfile cleanup

ormergi opened the pull request:
This PR cleans up Makefile a bit, it removes targets we currently dont need, such as building bundle images, catalog images, multi arch images.

All remaining targets are verified and work.

ormergi committed:
makefile: Remove bundle and catalog related targets

No need for these targets at the moment.

Signed-off-by: Or Mergi <ormergi@redhat.com>

ormergi committed:
makefile: Remove multi arch image build target and comments

No need to provide multu arch images at this point.

Signed-off-by: Or Mergi <ormergi@redhat.com>

ormergi committed:
makefile: Parameterize container runtime binary

Introduce CONTAINER_TOOL variable to enable controlling the container runtime
binary that will be used, following latest operator-sdk Makefile example.

The builder container image tag is changed to full form to make it work for
non docker container runtime.

Signed-off-by: Or Mergi <ormergi@redhat.com>

ormergi committed:
gitignore: Go cover profile output

Make test generates 'cover.out' file

Signed-off-by: Or Mergi <ormergi@redhat.com>

ormergi committed:
makefile: Remove envtest targets

No need to envtests at the moment.

Signed-off-by: Or Mergi <ormergi@redhat.com>

AlonaKaplan commented on Makefile:
```diff
@@ -4,6 +4,8 @@ IMAGE_TAG ?= devel
 # Image URL to use all building/pushing image targets
 IMG ?= "${IMAGE_REGISTRY}/${IMAGE_NAME}:${IMAGE_TAG}"
 
+OCI_BIN ?= podman
```
`OCI_BIN ?= $(shell if podman ps >/dev/null 2>&1; then echo podman; elif docker ps >/dev/null 2>&1; then echo docker; fi)` can be used to fins the oci bin.

ormergi committed:
cluster: Fix cluster-up/down targets

In case Kind is not installed, cluster-up and cluster-down target fails.
In addition, in case ./bin directory is not exist cluster-up fails.

Set cluster-up and cluster-down target to depend on LOCALBIN target.
Change cluster.sh to alway check if Kind binary exist.

Signed-off-by: Or Mergi <ormergi@redhat.com>

ormergi committed:
makefile: Use LOCALBIN target for image tar file path

Signed-off-by: Or Mergi <ormergi@redhat.com>

ormergi committed:
makefile: Parameterize kubectl binary

Introduce KUBECTL makefile variable to control kubectl binary path

Signed-off-by: Or Mergi <ormergi@redhat.com>
